x-pack/plugins/security_solution/server/types.ts

```diff
@@ -45,7 +45,7 @@ export interface SecuritySolutionApiRequestHandlerContext {
   getAppClient: () => AppClient;
   getSpaceId: () => string;
   getRuleDataService: () => IRuleDataService;
-  getDetectionRulesClient: () => IDetectionRulesClient;
+  getDetectionRulesClient: (actionsClient?: ActionsClient) => IDetectionRulesClient;
```

Comment:
Since now we're passing an initialized `actionsClient` to `createDetectionRulesClient`, you don't need to pass it as a dependency here.

Reply:
thank you yeah I still have cleanup left to do in this branch. I appreciate the early reviews though!